src/components/DuetRoom.tsx

```diff
@@ -73,21 +107,46 @@ const DuetRoom: React.FC<{ maybeRoomId: string | null; isCreate: boolean }> = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [maybeRoomId]);
 
+  const [middleBoxDimensions, middleBoxRef] = useDimensions<HTMLDivElement>();
+  // This is how to access the box dimensions.
+  // It changes dynamically when the window resizes.
+  /* TODO: remove this when doing waterfall */
+  console.log(middleBoxDimensions);
+
+  const friendId = getFriendId(roomState, playerId);
+  const partsSelection = getPartsSelection(roomState);
+
   return (
-    <>
-      <RoomContext.Provider
+    <RoomContext.Provider
+      value={{
+        roomInfo: roomState,
+        setRoomInfo: setRoomState,
+      }}
+    >
+      <PlayerContext.Provider
         value={{
-          roomInfo: roomState,
-          setRoomInfo: setRoomState,
+          me: playerId,
+          friend: friendId,
         }}
       >
-        <PlayerContext.Provider
-          value={{
-            me: playerId,
-            friend: getFriendId(roomState, playerId),
-          }}
-        >
-          <RoomHeader />
+        <Box className={classes.root}>
+          {/* header */}
+          <div className={classes.header}>
+            <RoomHeader />
+          </div>
+
+          {/* available space for the rest of the content */}
+          <div ref={middleBoxRef} className={classes.box}>
+            <PartSelection
```

Comment:
I am not very sure about the part selection logic so I would like to confirm that this space will be taken up by the waterfall some time (not necessarily immediately) after the part is selected. Is this right? 😅

Reply:
Yes! When the game starts the content of this div should be swapped out.